src/create-constructor-function.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Creates a constructor function from an ES2015 class, this is a workaround
+ * needed to being able to extend a class from code that's transpiled by Babel.
+ * See https://github.com/babel/babel/pull/8656
+ * @param {*} Type The ES2015 base class used to create the constructor
+ * @param {*} Parent The object where the prototype chain walk to copy over
+ * symbols and static properties to the constructor stops. If not provided,
+ * the chain stops at Object.
+ * @returns {Function} Constructor function than can be safely subclassed from
+ * transpiled code.
+ */
+function createConstructorFunction(Type, Parent) {
+	if (typeof Parent === "undefined") {
+		Parent = Object.getPrototypeOf(Object);
+	}
+
+	function TypeConstructor() {
+		return Reflect.construct(Type, arguments, this.constructor);
+	}
+
+	TypeConstructor.prototype = Object.create(Type.prototype);
+	TypeConstructor.prototype.constructor = TypeConstructor;
+
+	// Walk up the prototype chain to copy over all Symbols and
+	// static properties to the constructor function
+	let Link = Type;
+	while (Link !== Parent && Link !== null) {
+		const symbols = Object.getOwnPropertySymbols(Link);
+		const props = Object.getOwnPropertyNames(Link);
+
+		props.forEach(function(prop) {
+			if (!TypeConstructor[prop]) {
+				const value = Link[prop];
```

Comment:
did you try using `Object.getOwnPropertyDescriptor` instead? Doing so would allow you to also inherit the enumerability / configurability / writability of the property.

Reply:
I did not.

Reply:
Changed this, and added a test.